Prose/ProseLib/Sources/ProseCoreStub/StatusStore.swift

```diff
@@ -8,12 +8,43 @@
 import Foundation
 import SharedModels
 
+public struct StatusStore {
+    private var _onlineStatus: (_ jid: JID) -> OnlineStatus?
+    private var _lastSeenDate: (_ jid: JID) -> Date?
+    private var _timeZone: (_ jid: JID) -> TimeZone?
+    private var _statusLine: (_ jid: JID) -> (Character, String)?
+    
+    public func onlineStatus(for jid: JID) -> OnlineStatus? {
```

Comment:
The missing argument labels in closures are annoying indeed. You still have the named parameters at least for documentation purposes (`(_ jid: JID)`). Personally I think as long as we have strong and expressive types it's not too bad that we're loosing the argument labels.

Reply:
Hm, now we wouldn't be able to inject test code into the properties because they are immutable. Have you've worked with the TestStore before? What I'm hinting at is [this pattern](https://github.com/pointfreeco/swift-composable-architecture/blob/2bedb4b7aa8121730cd8d6db1bbff8304f8d03f1/Examples/TicTacToe/tic-tac-toe/Tests/TwoFactorCoreTests/TwoFactorCoreTests.swift#L17).

Reply:
I haven't yet worked with it (by lack of time), but I'm sure I saw it on pointfree.co someday. I didn't realize we would need to mutate those values, as we're not currently doing it (the compiler didn't complain), so I put `let` out of habit. I will fix this.